lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -215,8 +215,7 @@ def integration_message
         def create_or_update_build_phase(target, phase_name, phase_class = Xcodeproj::Project::Object::PBXShellScriptBuildPhase)
           prefixed_phase_name = BUILD_PHASE_PREFIX + phase_name
           build_phases = target.build_phases.grep(phase_class)
-          build_phases.find { |phase| phase.name == prefixed_phase_name } ||
-            build_phases.find { |phase| phase.name == phase_name }.tap { |p| p.name = prefixed_phase_name if p } ||
+          build_phases.find { |phase| phase.name && phase.name.end_with?(phase_name) }.tap { |p| p.name = prefixed_phase_name if p } ||
```

Comment:
Nope, if you remove that `bundle spec` crashes, that's why I had to add that security (that was the same for `if p` at the time I implemented the original feature).

The `if p` is necessary because if there is no build phase with that name at all, then it will `nil.tap { |p| … }` which _does_ call the block with `nil`, so that's a use case that can happen.

Not sure why I had to the add the same security for `phase_name &&` that time, didn't really try to understand why that check was now necessary (why `bundle spec` crashed with a `nil:Nil` exception) but better safe than sorry.